src/components/Trade/hooks/useSwapper/useSwapper.tsx

```diff
@@ -40,25 +38,35 @@ export const useSwapper = () => {
 
   // Selectors
   const flags = useSelector(selectFeatureFlags)
-  const assetIds = useSelector(selectAssetIds)
+  const assetsByMarketCapById = useSelector(selectSortedAssetsById)
 
   // Hooks
-  const [swapperManager, setSwapperManager] = useState<SwapperManager>(() => new SwapperManager())
+  const [swapperManager, setSwapperManager] = useState<SwapperManager>()
   const wallet = useWallet().state.wallet
 
-  // Callbacks
-  const getSupportedSellableAssets = useCallback(
-    (assets: Asset[]) => {
-      const sellableAssetIds = swapperManager.getSupportedSellableAssetIds({
-        assetIds,
-      })
+  // Selectors
+  const supportedSellAssetsByMarketCap = useMemo(() => {
+    if (!swapperManager) return []
 
-      return filterAssetsByIds(assets, sellableAssetIds)
-    },
-    [assetIds, swapperManager],
-  )
+    const sellableAssetIds = swapperManager.getSupportedSellableAssetIds({
+      assetIds: Array.from(assetsByMarketCapById.keys()),
+    })
+
+    return sellableAssetIds.map(assetId => assetsByMarketCapById.get(assetId)!)
+  }, [assetsByMarketCapById, swapperManager])
+
+  const supportedBuyAssetsByMaketCap = useMemo(() => {
+    const sellAssetId = sellAsset?.assetId
+    if (sellAssetId === undefined || !swapperManager) return []
+
+    const sellableAssetIds = swapperManager.getSupportedBuyAssetIdsFromSellId({
+      assetIds: Array.from(assetsByMarketCapById.keys()),
+      sellAssetId,
+    })
+
+    return sellableAssetIds.map(assetId => assetsByMarketCapById.get(assetId)!)
```

Comment:
Can we do without non-null assert?

Reply:
this is ok in this case as it's a filtered list of the known assets above